packages/tw-shimmer/src/index.css

```diff
@@ -51,3 +92,79 @@
 @utility shimmer-angle-* {
   --shimmer-angle: calc(--value(number) * 1deg);
 }
+
+@utility shimmer-bg {
+  --shimmer-color: oklch(1 0 0 / 0.5);
+  --shimmer-bg-spread: 400;
+  position: relative;
+  overflow: clip;
+
+  @variant dark {
+    --shimmer-color: oklch(1 0 0 / 0.25);
+  }
+
+  &::after {
+    content: "";
+    position: absolute;
+    top: -150%;
+    bottom: -150%;
+    left: 0;
+    width: calc(var(--shimmer-bg-spread) * 1px);
+    /* Eased gradient: sine ease-in-out for smooth falloff at both center and edges */
+    background: linear-gradient(
+      90deg,
+      transparent 0%,
+      color-mix(in oklch, var(--shimmer-color), transparent 96%) 6%,
+      color-mix(in oklch, var(--shimmer-color), transparent 83%) 13%,
+      color-mix(in oklch, var(--shimmer-color), transparent 67%) 19%,
+      color-mix(in oklch, var(--shimmer-color), transparent 50%) 25%,
+      color-mix(in oklch, var(--shimmer-color), transparent 33%) 31%,
+      color-mix(in oklch, var(--shimmer-color), transparent 17%) 38%,
+      color-mix(in oklch, var(--shimmer-color), transparent 4%) 44%,
+      var(--shimmer-color) 50%,
+      color-mix(in oklch, var(--shimmer-color), transparent 4%) 56%,
+      color-mix(in oklch, var(--shimmer-color), transparent 17%) 62%,
+      color-mix(in oklch, var(--shimmer-color), transparent 33%) 69%,
+      color-mix(in oklch, var(--shimmer-color), transparent 50%) 75%,
+      color-mix(in oklch, var(--shimmer-color), transparent 67%) 81%,
+      color-mix(in oklch, var(--shimmer-color), transparent 83%) 87%,
+      color-mix(in oklch, var(--shimmer-color), transparent 96%) 94%,
+      transparent 100%
+    );
+    transform: skewX(calc(90deg - var(--shimmer-angle, 90deg)));
+    animation: shimmer-bg
+      calc(
+        (var(--shimmer-width, 600) + var(--shimmer-bg-spread)) /
+          var(--shimmer-speed, 500) * 1s
+      )
+      infinite linear;
+    /* Delay syncs animation across elements: y adjusted by angle slope, minus x offset */
+    animation-delay: calc(
+      (
+          var(--shimmer-y, 0) * tan(calc(90deg - var(--shimmer-angle, 90deg))) -
+            var(--shimmer-x, 0)
+        ) /
+        var(--shimmer-speed, 500) * -1s
+    );
+  }
+}
+
+@utility shimmer-width-* {
+  --shimmer-width: --value(number);
+}
+
+@utility shimmer-y-* {
```

Comment:
See my comment about the anchor() implementation that would remove the need to.manually set the x/y -- then most users would only need to set the shimmer-container class and not think about x/y

IMO the utilities for x/y can be skipped (instead rely on explicit CSS var setting if the user wants to set it manually)